test/OpenTelemetry.Instrumentation.GrpcCore.Tests/GrpcCoreServerInterceptorTests.cs

```diff
@@ -161,6 +161,6 @@ private static async Task TestHandlerFailure(Func<Foobar.FoobarClient, Metadata,
 
         var activity = activityListener.Activity;
         GrpcCoreClientInterceptorTests.ValidateCommonActivityTags(activity, StatusCode.ResourceExhausted, interceptorOptions.RecordMessageEvents, interceptorOptions.RecordException);
-        Assert.Equal(FoobarService.DefaultParentFromTraceparentHeader.SpanId, activity.ParentSpanId);
+        Assert.Equal(FoobarService.DefaultParentFromTraceparentHeader.SpanId, activity!.ParentSpanId);
```

Comment:
Consider, it applies in other places where you put `!` is tests.

```suggestion
        Assert.NotNull(activity);
        Assert.Equal(FoobarService.DefaultParentFromTraceparentHeader.SpanId, activity.ParentSpanId);
```

Reply:
The idea is that `ValidateCommonActivityTags` ensures that `activity` is not null. Added `[NotNull]` attribute for the activity argument to the method. So, `!` is no longer needed